__tests__/frontend-proxy.ts

```diff
@@ -0,0 +1,215 @@
+      const res = await handleUrl(testUrl, 1)
+      expect(res).not.toBe(frontendResponse)
+    })
+  })
+
+  describe('transfers request meta data to backend', () => {
+    test.each([true, false])('useFrontend=%p', async (useFrontend) => {
+      await withMockedFetch(
+        [createApiResponse(), checkRequestData],
+        async () => {
+          const cookie = `useFrontend20=${useFrontend};`
+          const response = await handleUrl(testUrl, 0.2, cookie, {
+            headers: { 'X-Header': 'foo' },
+          })
+
+          expect(await response.text()).toBe('header-shown')
+        }
+      )
+
+      function checkRequestData(req: Request) {
+        const res =
+          req.headers.get('X-Header') === 'foo'
+            ? new Response('header-shown')
+            : new Response('no-header')
+
+        return new Promise<Response>((resolve) => resolve(res))
+      }
+    })
+  })
+
+  describe('transfers response meta data from backend', () => {
+    test.each([true, false])('useFrontend=%p', async (useFrontend) => {
+      const targetResponse = new Response('', {
+        headers: { 'X-Header': 'bar' },
+      })
+
+      await withMockedFetch([createApiResponse(), targetResponse], async () => {
+        const cookie = `useFrontend20=${useFrontend};`
+        const res = await handleUrl(testUrl, 0.2, cookie)
+
+        expect(res.headers.get('X-Header')).toBe('bar')
+      })
+    })
+  })
+
+  test('requests to /enable-frontend enable use of frontend', async () => {
+    const url = 'https://de.serlo.org/enable-frontend'
+    const res = await handleUrl(url, 0.15)
+
+    hasOkStatus(res)
+    expect(res.headers.get('Set-Cookie')).toBe('useFrontend15=true; path=/')
+    expect(await res.text()).toBe('Enable frontend')
+  })
+
+  test('requests to /disable-frontend disable use of frontend', async () => {
+    const url = 'https://de.serlo.org/disable-frontend'
+    const res = await handleUrl(url, 0.15)
+
+    hasOkStatus(res)
+    expect(res.headers.get('Set-Cookie')).toBe('useFrontend15=false; path=/')
+    expect(await res.text()).toBe('Disable frontend')
+  })
+})
+
+test('createApiQuery()', () => {
+  expect(createApiQuery('/math')).toBe(
+    '{ uuid(alias: { instance: de, path: "/math" }) { __typename } }'
+  )
+  expect(createApiQuery('hello')).toBe(
+    '{ uuid(alias: { instance: de, path: "/hello" }) { __typename } }'
+  )
+  expect(createApiQuery('/266/')).toBe(
+    '{ uuid(alias: { instance: de, path: "/266/" }) { __typename } }'
+  )
+  expect(createApiQuery('/266')).toBe('{ uuid(id: 266) { __typename } }')
+  expect(createApiQuery('/874329')).toBe('{ uuid(id: 874329) { __typename } }')
+})
+
+async function handleUrl(
+  url: string,
+  probability = 0.1,
+  cookie?: string,
+  reqInit?: RequestInit
+) {
+  const req = new Request(url, reqInit)
+
+  if (cookie !== undefined) req.headers.set('Cookie', cookie)
+
+  return (await handleRequest(req, probability, TEST_ALLOWED_TYPES)) as Response
+}
```

Comment:
Even if you want to use this helper, this should be split into multiple ones. It contains two flag arguments (`cookie` and `reqInit`) that change the assumptions of the test. I'd probably remove the default argument from `probability`, too, since I'd like that to see in the test instead.